### ISSUE TITLE
Array to float should emit an error

### ISSUE THREAD

**Issue by oleg-andreyev** [vimeo/psalm#7947]
```php
<?php

$data = ['price' => '43.444444444444444'];
echo round((float) $data, 4);
```

since $data is not empty, it will converted to 1.00, to see an error

https://psalm.dev/r/6b22edfc6e

**Comment by AndrolGenhald** [comment]
Since it's an explicit cast I'm not sure we should show an error for that, but perhaps we could have a setting to be more strict about certain casts? We'd have to think it through and decide what casts should and shouldn't be allowed.

**Comment by weirdan** [comment]
Now emits `RiskyCast`: https://psalm.dev/r/6b22edfc6e
